br/pkg/lightning/backend/external/byte_reader.go

```diff
@@ -33,6 +38,12 @@ type byteReader struct {
 	bufOffset int
 
 	retPointers []*[]byte
+
+	useConcurrentReaderCurrent atomic.Bool
+	useConcurrentReader        atomic.Bool
+
+	currFileOffset int64
+	conReader      *singeFileReader
```

Comment:
If time permits, I prefer we unify `storageReader + buf + bufOffset` and `conReader` to one interface. So we don't need another if-branch to switch from which reader we will read